store.js

```diff
@@ -0,0 +1,283 @@
+const { pack, unpack }      = require('msgpackr')
+const { mkdirSync }         = require('fs')
+const { Cursor, Env }       = require('node-gyp-build')(__dirname)
+const { AsyncLocalStorage } = require('async_hooks')
+
+function asBinary(buffer) {
+  return {
+    ['\x10binary-data\x02']: buffer,
+  }
+}
+
+function startsWithBuffer(buffer, prefix) {
+  if (buffer.length < prefix.length) {
+    return false
+  }
+
+  for (let i = 0; i < prefix.length; i++) {
+    if (buffer[i] !== prefix[i]) {
+      return false
+    }
+  }
+
+  return true
+}
+
+
+class Iterator {
+  constructor(env, dbi, options = {}) {
+    this.txn = env.beginTxn({ readOnly: true })
+    this.cursor = new Cursor(this.txn, dbi)
+    this.options = options
+    this.prefixBuffer = options.prefix ? Buffer.from(options.prefix) : null
+  }
+
+  close() {
+    this.cursor.close()
+    this.txn.abort() // a readonly txn
+  }
+
+  [Symbol.iterator]() {
+    const self = this
+    const { keys = true, values = true } = this.options
+    let key = this.prefixBuffer ? this.cursor.goToRange(this.prefixBuffer) : this.cursor.goToFirst()
+
+    return {
+      next() {
+        if (key !== null && (!self.prefixBuffer || startsWithBuffer(key, self.prefixBuffer))) {
+          let value
+          if (keys && values) {
+            value = {
+              key: key.toString('utf8'),
+              value: unpack(self.cursor.getCurrentBinary())
+            }
+          } else if (keys) {
+            value = key.toString('utf8')
+          } else if (values) {
+            value = unpack(self.cursor.getCurrentBinary())
+          } else {
+            value = null
+          }
+          key = self.cursor.goToNext()
+          return { value, done: false }
+        } else {
+          self.close()
+          return { done: true }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * This class enforces a very specific usage of lmdb. All keys are
+ * utf8 encoded buffers and all values are encoded with
+ * msgpack. Transaction handling is also intentionally simplified to
+ * just one current transaction, no nesting. This is the pattern that
+ * adset-consumer uses in its current form.
```

Comment:
Added a note about only supporting one named DB since that is another piece of functionality that has been closed over

```suggestion
 * This class enforces a very specific usage of lmdb. All keys are
 * utf8 encoded buffers and all values are encoded with
 * msgpack. Only one named dbi is supported. Transaction 
 * handling is also intentionally simplified to
 * just one current transaction, no nesting. This is the pattern that
 * adset-consumer uses in its current form.
```